lib/core/test/unit/Cardano/Wallet/Primitive/Types/UTxOSpec.hs

```diff
@@ -50,6 +61,10 @@ spec :: Spec
 spec =
     describe "Cardano.Wallet.Primitive.Types.UTxOSpec" $ do
 
+    parallel $ describe "delta encoding" $ do
+        it "DeltaUTXO is a semigroup" $
+            property prop_deltaUTxO_semigroup
```

Comment:
I wonder using `quickcheck-classes` here would be helpful?

For example: https://github.com/input-output-hk/cardano-wallet/blob/3ecad89bbcc4c4baa70e3cb4b5695aecdb0d92f5/lib/core/test/unit/Cardano/Wallet/Primitive/Types/TokenMapSpec.hs#L129

Reply:
Not necessarily, for two reasons:

* We not only want to check that `DeltaUTxO` is a monoid, but also that this monoid is compatible with `apply`. The latter property is more important, and actually implies that the monoid laws hold (assuming that `apply` is the only "observable"). In other words, the real test is whether `apply` is a monoid morphism.
* When checking the associativity law `(a <> b) <> c = a <> (b <> c)`, the values `a`,`b`, and `c` are generated randomly. But with the standard generators, it is exceedingly unlikely that the values contain overlapping `TxIn`. In this case, the `Semigroup` instance would be indistinguishable from the *wrong* instance
  ```
  instance Semigroup DeltaUTxO where
      db <> da = DeltaUTxO
          { excluded = excluded da <> excluded db -- wrong
          , received = received da <> received db -- nope
          }
  ```
    (I didn't try this, but it looks like an interesting exercise.)
    Instead of fiddling with generators to generate overlapping values, I might as well write down the complete Venn diagram; this relies on the assumption that the implementation of `DeltaUTxO` is implemented only in terms of `Set` and `Map` and thus can be analyzed in terms of a Venn diagram.